src/main/java/seedu/storage/Storage.java

```diff
@@ -38,4 +72,22 @@ public ContactList loadExistingContacts() throws FileErrorException {
         }
         return ContactsDecoder.readContacts(contactFile);
     }
+
+    public Contact loadExistingPersonalContact() throws FileErrorException {
+        if (!hasExistingPersonalContactFile() || hasEmptyExistingPersonalContactFile()) {
+            isFirstRun = true;
+            // get new contact's name
+            TextUi.welcomeMessage();
+            String personalName = UserInputTextUi.getUserInput();
+            Contact personalContact = new Contact(personalName, null,null,null,
+                    null,null);
```

Comment:
Hi @lezongmun , I think a `PersonalContactParser` would be better as it can make use of the regex checkers. This can extend for other fields as well.